common/tier4_autoware_utils/include/tier4_autoware_utils/ros/published_time_publisher.hpp

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TIER4_AUTOWARE_UTILS__ROS__PUBLISHED_TIME_PUBLISHER_HPP_
+#define TIER4_AUTOWARE_UTILS__ROS__PUBLISHED_TIME_PUBLISHER_HPP_
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <autoware_internal_msgs/msg/published_time.hpp>
+#include <std_msgs/msg/header.hpp>
+
+#include <functional>
+#include <map>
+#include <string>
+
+namespace tier4_autoware_utils
+{
+using autoware_internal_msgs::msg::PublishedTime;
+
+// Custom comparison struct for rmw_gid_t
+struct GidCompare
+{
+  bool operator()(const rmw_gid_t & lhs, const rmw_gid_t & rhs) const
+  {
+    return std::memcmp(lhs.data, rhs.data, RMW_GID_STORAGE_SIZE) < 0;
+  }
+};
+
+class PublishedTimePublisher
+{
+public:
+  explicit PublishedTimePublisher(
+    rclcpp::Node * node, const std::string & end_name = "/debug/published_time",
+    const rclcpp::QoS & qos = rclcpp::QoS(1))
+  : node_(node), end_name_(end_name), qos_(qos)
+  {
+  }
+
+  void publish(const rclcpp::PublisherBase::ConstSharedPtr & publisher, const rclcpp::Time & stamp)
+  {
+    const auto & gid_key = publisher->get_gid();
+    const auto & topic_name = publisher->get_topic_name();
+
+    // if the publisher is not in the map, create a new publisher for published time
+    ensure_publisher_exists(gid_key, topic_name);
```

Comment:
```suggestion

    // if the publisher is not in the map, create a new publisher for published time
    ensure_publisher_exists(gid_key, publisher->get_topic_name());
```

same as https://github.com/autowarefoundation/autoware.universe/pull/6440/files#r1519499450